lms/djangoapps/dashboard/git_import.py

```diff
@@ -293,7 +298,20 @@ def add_repo(repo, rdir_in, branch=None):
     match = re.search(r'(?ms)===> IMPORTING courselike (\S+)', ret_import)
     if match:
         course_id = match.group(1)
-        course_key = CourseKey.from_string(course_id)
+        try:
+            course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError:
+            course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+
+        try:
+            get_course_by_id(course_key)
+        except Exception:
```

Comment:
Instead of nesting these exceptions, could you coerce the course_id to the CourseLocator object around line 302 and then simply make the call please